test/test_invalid_headers.py

```diff
@@ -95,30 +97,49 @@ class TestFilter(object):
     HTTP/2 and so may never hit the function, but it's worth validating that it
     behaves as expected anyway.
     """
-    true_false_combinations = [
-        (True, True),
-        (True, False),
-        (False, True),
-        (False, False),
+    validation_functions = [
+        h2.utilities.validate_headers,
+        h2.utilities.validate_sent_headers
     ]
+    validator_combinations = list(itertools.product(
+        validation_functions,
+        (True, False),
+        (True, False),
```

Comment:
Notes to self:

1. Do you definitely need to cast to a list?
2. Better if this had the logic for constructing header validation flags, and those were passed directly into the tests; hoist a bit more logic out of the tests themselves.